src/modules/ChannelList/dux/reducers.js

```diff
@@ -114,79 +122,94 @@ export default function reducer(state, action) {
     }
     case actions.ON_USER_LEFT: {
       const { channel, isMe } = action.payload;
-      if (state.channelListQuery) {
-        if (filterChannelListParams(state.channelListQuery, channel, state.currentUserId)) {
-          const filteredChannels = getChannelsWithUpsertedChannel(state.allChannels, channel);
-          const nextChannel = (isMe && (channel?.url === state.currentChannel?.url))
-            ? filteredChannels[0]
-            : state.currentChannel;
-          return {
-            ...state,
-            currentChannel: state.disableAutoSelect ? null : nextChannel,
-            allChannels: filteredChannels,
-          };
+      const {
+        allChannels,
+        currentUserId,
+        currentChannel,
+        channelListQuery,
+        disableAutoSelect,
+      } = state;
+      let nextChannels = allChannels.filter((ch) => ch.url !== channel.url);
+      let nextChannel = null;
+      if (channelListQuery) {
+        if (filterChannelListParams(channelListQuery, channel, currentUserId)) {
+          // Good to [add to/keep in] the ChannelList
+          nextChannels = getChannelsWithUpsertedChannel(allChannels, channel);
         }
-        const nextChannel = (channel?.url === state.currentChannel?.url)
-          ? state.allChannels[0]
-          : state.currentChannel;
-        return {
-          ...state,
-          currentChannel: state.disableAutoSelect ? null : nextChannel,
-          allChannels: state.allChannels.filter(({ url }) => url !== channel?.url),
-        };
       }
-      const filteredChannels = state.allChannels.filter((c) => !(c.url === channel?.url && isMe));
-      const nextChannel = (isMe && (channel?.url === state.currentChannel?.url))
-        ? filteredChannels[0]
-        : state.currentChannel;
+      // Replace the currentChannel if I left the currentChannel
+      if (isMe && channel.url === currentChannel?.url) {
```

Comment:
It seems the similar logic is in the `getNextChannel` util function

Reply:
Yes there is, but `isMe` is added, so little bit different

Reply:
She is right. It is redundant code. Let's replace with
```
let nextChannel;
if (isMe) {
  nextChannel = getNextChannel();
}
```